File: Lab3/C#/StringLines.cs
using System;

namespace Lab3
{
    public class Line
    {
        private string value;

        public Line()
        {
            value = "DEFAULT";
        }

        public Line(string input)
        {
            value = input;
        }

        public Line(Line copiedObj)
        {
            value = copiedObj.value;
        }

        public bool IsUppercaseLatin()
        {
            return value.All(c => c >= 'A' && c <= 'Z');
        }

        public string GetValue()
        {
            return value;
        }

        public int GetLength()
        {
            return value.Length;
        }


        public static Line operator +(Line a, Line b)
        {
            return new Line(a.GetValue() + b.GetValue());
        }

        public static Line operator /(Line s, int delimiter)
        {
            string newValue = "";
            for (int i = 0; i < s.value.Length; i++)
            {
                if (i % delimiter == 0)
                {
                    newValue += s.value[i];
                }
            }
            return new Line(newValue);
        }

        // public void DeleteEvenChars(string input)
        // {
        //     string result = "";
        //     for (int i = 0; i < input.Length; i++)
        //     {
        //         if (i % 2 != 0)
        //         {
        //             result += input[i];
        //         }
        //     }
        //     value = result;
        // }

        // ~Line()
        // {
        //     Console.WriteLine("object destroyed");
        // }
    }
}

File: Lab3/C++/stringLines.cpp
#include "stringLines.h"
#include <algorithm>
#include <iostream>

Lines::Lines()
{
    value = "default";
}

Lines::Lines(std::string input) 
{
    value = input;
}

Lines::Lines(const Lines &copied)
{
    value = copied.value;
}

bool Lines::isValidString(std::string input)
{
    for (char c : input)
    {
        if (c < 'A' || c > 'Z')
        {
            return false;
        }
    }
    return true;
}

std::string Lines::getText()
{
    return value;
}

size_t Lines::stringLength()
{
    return value.length();
}


Lines Lines::operator/(const size_t delimiter) const
{
    std::string newValue = "";
    for (size_t i = 0; i < value.length(); i++)
    {
        if (i % delimiter == 0)
        {
            newValue += value[i];
        }
    }
    return Lines(newValue);
}

Lines Lines::operator+(const Lines &other) const
{
    std::string result = value + other.value;
    return Lines(result);
}

// Lines::~Lines()
// {
//     std::cout << "\033[36m>> destructor called" << std::endl;
// }
